nanoFramework.Tools.DebugLibrary.Shared/PortComposite/PortCompositeDeviceManager.cs

```diff
@@ -52,21 +49,30 @@ private void OnLogMessageAvailable(object sender, StringEventArgs e)
 
         private void OnPortDeviceEnumerationCompleted(object sender, EventArgs e)
         {
-            DeviceEnumerationCompleted?.Invoke(this, EventArgs.Empty);
+            IsDevicesEnumerationComplete = (from p in _ports
+                                            where p.IsDevicesEnumerationComplete
+                                            select p).Any();
+            if (IsDevicesEnumerationComplete)
+            {
+                DeviceEnumerationCompleted?.Invoke(this, EventArgs.Empty);
+            }
         }
 
         /// <inheritdoc/>
         /// <exception cref="NotImplementedException">This API is not available in </exception>
-        public override void AddDevice(string deviceId)
+        public override NanoDeviceBase AddDevice(string deviceId)
```

Comment:
Ok, added a new method (in the next commit) that returns the device. (use case for the method: the software of the test framework v3 uses AddDevice to get access to the device, but then has to search the global list to find the resulting device. It really helps to return the device added).

Reply:
I wasn't aware of the need to return the device, that's what I meant with my initial comment.
If that is needed, by means let's have it. No need to have "duplicated" APIs doing pretty much the same thing.

(It's OK to clarify the reasoning behind what was done before reverting it or come up with an alternative. Better do that and wait for a final resolution before resolving the conversation. This is an iterative process. Plus, it can save time and not waste work. 😉 )

Reply:
Ok, I'll go back to AddDevice with return value.

Looking back to the description I didn't mention this. But it is in one of the few hundred lines of descriptions of changes in the test framework. You don't remember? :-)

In the test framework it is possible to specify which port should be used. The code then creates a PortSerialManager and calls AddDevice with the serial port as argument. If the current AddDevice returns, the test framework does not know the device and does not even know whether it is added. It then has to enumerate the global list of NanoDevices to find the device.

(I'm not very used to PR with online discussions)

Reply:
> Ok, I'll go back to AddDevice with return value.

Please do. It's now clear the need and motivation.

 
> Looking back to the description I didn't mention this. But it is in one of the few hundred lines of descriptions of changes in the test framework. You don't remember? :-)

🤪 I'm afraid not... you'll forgive my poor memory... I keep overloading it with too many matters simultaneouly. 

 
> In the test framework it is possible to specify which port should be used. The code then creates a PortSerialManager and calls AddDevice with the serial port as argument. If the current AddDevice returns, the test framework does not know the device and does not even know whether it is added. It then has to enumerate the global list of NanoDevices to find the device.

Sure! That API was already being used in v2, just that there was no need to get the device added. All good.

 
> (I'm not very used to PR with online discussions)

No worries! You're doing fine. This PR etiquette is something that one gets use to very quickly. Basically, keep the communication flowing, make sure everyone is in the same page, decide on something, then act. Trying not to waste anyone's time and be efficient.